File: LocalPayment/src/main/java/com/braintreepayments/api/LocalPaymentAuthResult.java
package com.braintreepayments.api;

import android.content.Context;

/**
 * Result received from the local payment web flow through
 * {@link LocalPaymentInternalTokenizeCallback}. This result should be passed to
 * {@link LocalPaymentClient#tokenize(Context, LocalPaymentAuthResult, LocalPaymentInternalTokenizeCallback)}
 * to complete the local payment flow.
 */
public class LocalPaymentAuthResult {

    private BrowserSwitchResult browserSwitchResult;
    private Exception error;

    LocalPaymentAuthResult(BrowserSwitchResult browserSwitchResult) {
        this.browserSwitchResult = browserSwitchResult;
    }

    LocalPaymentAuthResult(Exception error) {
        this.error = error;
    }

    BrowserSwitchResult getBrowserSwitchResult() {
        return browserSwitchResult;
    }

    Exception getError() {
        return error;
    }

}

File: LocalPayment/src/main/java/com/braintreepayments/api/localpayment/LocalPaymentAuthResult.kt
package com.braintreepayments.api.localpayment

import com.braintreepayments.api.BrowserSwitchFinalResult

/**
 * Result of the local payment web flow received from [LocalPaymentLauncher.handleReturnToAppFromBrowser].
 */
sealed class LocalPaymentAuthResult {

    /**
     * A successful result that should be passed to [LocalPaymentClient.tokenize] to complete the flow
     */
    class Success internal constructor(
        internal val browserSwitchSuccess: BrowserSwitchFinalResult.Success
    ) : LocalPaymentAuthResult()

    /**
     * The browser switch failed.
     * @property [error] Error detailing the reason for the browser switch failure.
     */
    class Failure internal constructor(val error: Exception) : LocalPaymentAuthResult()

    /**
     * If no matching result can be found for the [LocalPaymentPendingRequest.Started] passed to
     * [LocalPaymentLauncher.handleReturnToAppFromBrowser]. This is expected if the user closed the
     * browser to cancel the payment flow, or returns to the app without completing the
     * authentication flow.
     */
    data object NoResult : LocalPaymentAuthResult()
}
